var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -246,6 +248,15 @@ class Mfem(Package):
     depends_on('umpire@2.0.0:', when='+umpire')
     depends_on('umpire+cuda', when='+umpire+cuda')
 
+    depends_on('amgx', when='+amgx')
+    # AmgX doesn't need CUDA but MFEM build system requires amgx+cuda
+    depends_on('cuda', when='+amgx')
```

Comment:
It is better to express this as a conflict: `+amgx~cuda` is a conflict. Otherwise you probably end up with a cuda build where the `cuda` variant is off: `~cuda` in the MFEM spec.